gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,949 @@
+    "            print('Sample workspace contains one histogram.')\n",
+    "            print(f'Elastic Workspace contains {wavelength[\"sample_elastic\"].coords[\"spectrum\"].shape[0]} histograms.')\n",
+    "            #Fitting the masked sample\n",
+    "\n",
+    "\n",
+    "            sample_fit = sc.compat.mantid.fit(sample_ws,\n",
+    "                                          function=f'name=LinearBackground,A0={230},A1={-4};name=UserFunction,Formula=h*erf(a*(x-x0)),h={16},a={-11},x0={xpos_guess}',\n",
+    "                                          workspace_index=0, start_x=x_min_fit, end_x=x_max_fit)\n",
+    "\n",
+    "            params = dict(zip(sample_fit[\"parameters\"].value[\"Name\"].values, sample_fit[\"parameters\"].value[\"Value\"].values))\n",
+    "\n",
+    "            # take the fitting parameters and pass them on to the forthcoming fits\n",
+    "            d_sample = params[\"f1.x0\"] / 2.0 # See fit table definition for extract x0\n",
+    "            fit_values = (params[\"f0.A0\"], params[\"f0.A1\"], params[\"f1.h\"], params[\"f1.a\"], params[\"f1.x0\"])\n",
+    "\n",
+    "\n",
+    "        plots_counter = 0 # sets the workspace index to zero then increases one by one\n",
+    "        for row in range(y_size):\n",
+    "            for col in range(x_size):\n",
+    "                ws_index = plots[plots_counter][0]\n",
+    "\n",
+    "                # if the inital sample was grouped\n",
+    "                if \"spectrum\" in rebinned_wavelength[\"sample\"].coords:\n",
+    "\n",
+    "                    sample_fit = sc.compat.mantid.fit(sample_ws,\n",
+    "                                                  function=f\"name=LinearBackground,A0={230},A1={-4};name=UserFunction,Formula=h*erf(a*(x-x0)),h={16},a={-11},x0={xpos_guess}\",\n",
+    "                                                  workspace_index=ws_index, start_x=x_min_fit, end_x=x_max_fit)\n",
+    "                    \n",
+    "                    # take the fitting parameters and pass them on to the forthcoming fits\n",
+    "                    params = dict(zip(sample_fit[\"parameters\"].value[\"Name\"].values, sample_fit[\"parameters\"].value[\"Value\"].values))\n",
+    "\n",
+    "                    # take the fitting parameters and pass them on to the forthcoming fits\n",
+    "                    d_sample = params[\"f1.x0\"] # See fit table definition for extract x0\n",
+    "                    fit_values = (params[\"f0.A0\"], params[\"f0.A1\"], params[\"f1.h\"], params[\"f1.a\"], params[\"f1.x0\"])\n",
+    "\n",
+    "                # Fit Bragg edge using values from fit of unstrained sample\n",
+    "                elastic_function = f\"name=LinearBackground,A0={fit_values[0]},A1={fit_values[1]};name=UserFunction,Formula=h*erf(a*(x-x0)),h={fit_values[2]},a={fit_values[3]},x0={fit_values[4]}\"\n",
+    "                fit_elastic = sc.compat.mantid.fit(elastic_ws,\n",
```

Comment:
Seems strange that this `fit function takes a workspace as input`. Can we create an overload that internally does the dataset->workspace conversion that you apply apove with `convert_var_to_workpace`?